Userland/Libraries/LibCrypt/crypt.cpp

```diff
@@ -52,8 +52,13 @@ char* crypt_r(char const* key, char const* salt, struct crypt_data* data)
     sha.update(reinterpret_cast<u8 const*>(salt_value), salt_len);
 
     auto digest = sha.digest();
-    auto string = encode_base64(ReadonlyBytes(digest.immutable_data(), digest.data_length()));
+    auto string_or_error = encode_base64({ digest.immutable_data(), digest.data_length() });
+    if (string_or_error.is_error()) {
+        errno = ENOMEM;
+        return nullptr;
+    }
 
+    auto string = string_or_error.release_value().bytes_as_string_view();
```

Comment:
this smells like use after free. If you use `.value()` instead, it won't be.

Reply:
Good point, fixed!